Controller/Adminhtml/Menu/Export.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Snowdog\Menu\Controller\Adminhtml\Menu;
+
+use Magento\Backend\App\Action;
+use Magento\Framework\App\Filesystem\DirectoryList;
+use Magento\Framework\App\Response\Http\FileFactory as HttpFileFactory;
+use Psr\Log\LoggerInterface;
+use Snowdog\Menu\Model\Menu\ExportProcessor;
+
+class Export extends Action
+{
+    /**
+     * @inheritDoc
+     */
+    const ADMIN_RESOURCE = 'Snowdog_Menu::menus';
+
+    const FILE_NAME = 'menu-{menu_id}.csv';
+
+    /**
+     * @var HttpFileFactory
+     */
+    private $httpFileFactory;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var ExportProcessor
+     */
+    private $exportProcessor;
+
+    public function __construct(
+        Action\Context $context,
+        HttpFileFactory $httpFileFactory,
+        LoggerInterface $logger,
+        ExportProcessor $exportProcessor
+    ) {
+        $this->httpFileFactory = $httpFileFactory;
+        $this->logger = $logger;
+        $this->exportProcessor = $exportProcessor;
+
+        parent::__construct($context);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function execute()
```

Comment:
Do we really want to store all exports as downloadable files?
This causes:
* mess in `var`
* potential problems with writing files and permissions
* pseudorandom suffix in file name may be not sufficient for providing uniqueness

Reply:
Export files do not get permanently stored, they get temporarily stored to obtain the download file content.  
Before the download file response is returned, the file will get deleted. ([Check HTTP File Response `rm` flag](https://github.com/SnowdogApps/magento2-menu/blob/cc67abd08368843130b186fe40fc67548a70fcdd/Model/Menu/ExportProcessor.php#L100))  
That's how Magento HTTP file download response class [works](https://github.com/magento/magento2/blob/2.4.1/lib/internal/Magento/Framework/App/Response/Http/FileFactory.php#L107-L109).

I opted to use the ready [Magento file download](https://github.com/magento/magento2/blob/2.4.1/lib/internal/Magento/Framework/App/Response/Http/FileFactory.php) solution rather than writing the file download logic in the module,  
to keep the code as simple as possible, but with the caveat that export files will get temporarily stored before being downloaded.

The same goes for the menu import processor, which uses Magento ImportExport module [CSV file upload and data processing](https://github.com/magento/magento2/blob/2.4.1/app/code/Magento/ImportExport/Model/Import.php#L612-L630) feature.  
The temporary import file [gets deleted](https://github.com/SnowdogApps/magento2-menu/blob/cc67abd08368843130b186fe40fc67548a70fcdd/Model/Menu/ImportProcessor.php#L88) after obtaining the processed upload file data.